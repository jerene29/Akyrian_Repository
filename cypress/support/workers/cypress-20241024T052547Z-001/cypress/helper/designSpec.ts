import { GetStudyRevisionListDocument } from '../../src/graphQL/generated/graphql';
import { mockUserDataAdmin } from '../../src/constant/testFixtures';

type SpecNumber = `S-${number}-${number}`;

type SpecInfo = {
  // e.g S-09-010
  spec: SpecNumber;
  // title e.g headerSection, combined description becomes S-09-010 headerSection, but optional
  title?: string;
};

type Options = {
  loginAs?: string;
  skipTest?: boolean;
};

function resolveDescription(description: SpecNumber | SpecInfo) {
  return typeof description === 'string' ? description : `${description.spec} ${description.title}`;
}

const aliasGetStudyRevisionListDocument =
  'name' in GetStudyRevisionListDocument.definitions[0]
    ? GetStudyRevisionListDocument.definitions[0].name?.value
    : 'GetStudyRevisionListDocument';
/*
example usage
designSpec('S-19-2', () => {

  describe('...', ...)

  it('checks something...', () => {...})
});
*/
export function designSpec(
  description: SpecNumber | SpecInfo,
  callback: () => void,
  options?: Options,
) {
  before(() => {
    cy.intercept('POST', '/graphql', (req) => {
      if (req.body.operationName === aliasGetStudyRevisionListDocument) {
        req.alias = req.body.operationName;
      }
    });
    cy.beforeSetup(mockUserDataAdmin);
    cy.clearLocalStorageSnapshot();
    cy.saveLocalStorage();
    cy.wait(`@${aliasGetStudyRevisionListDocument}`);
  });
  if (options?.skipTest) {
    describe.skip(resolveDescription(description), callback);
  } else if (typeof description !== 'string' && description.title) {
    describe(description.spec, () => {
      describe(String(description.title), callback);
    });
  } else {
    describe(resolveDescription(description), callback);
  }
}
