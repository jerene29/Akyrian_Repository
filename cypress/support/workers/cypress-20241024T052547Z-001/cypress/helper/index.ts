export * from './designSpec';
export * from './object';