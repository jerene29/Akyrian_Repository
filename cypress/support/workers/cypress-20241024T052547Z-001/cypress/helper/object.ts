/**
 * to shorten the tedious '[data-cy=selector]' selector.
 * just put the name of data-cy attributed element.
 * -- d stands for document
 *
```
d`addButton`
```
 */
export function d(args: TemplateStringsArray) {
  return `[data-cy=${args.join('')}]`;
}
