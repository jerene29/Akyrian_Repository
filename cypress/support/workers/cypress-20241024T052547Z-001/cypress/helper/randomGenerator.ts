import moment = require("moment");

export const randomAlphaNumeric = (length: number) => {
  const chars = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
  let result = '';
  for (let i = length; i > 0; --i) result += chars[Math.floor(Math.random() * chars.length)];
  return result;
}

export const randomAlphabet = (length: number) => {
  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
  let result = '';
  for (let i = length; i > 0; --i) result += chars[Math.floor(Math.random() * chars.length)];
  return result;
}

export const randomDate = () => {
  const start = new Date(2012, 0, 1)
  const end = new Date()
  const date = new Date(start.getTime() + Math.random() * (end.getTime() - start.getTime()))
  return moment(date);
}