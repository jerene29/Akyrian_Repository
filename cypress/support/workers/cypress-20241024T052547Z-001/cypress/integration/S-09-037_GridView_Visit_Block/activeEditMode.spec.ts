import { d, designSpec } from '../../helper';

const options = {
  loginAs: 'admin@example.com',
};

designSpec(
  {
    spec: 'S-09-037',
    title: 'Active Edit Mode on Visit',
  },
  () => {
    before(() => {
      cy.createStudy({
        name: 'GRIDVIEW_TEST',
        formCount: 5,
        visitCount: 5,
      });
    });

    context('keyboard shortcut', () => {
      it('uses [RETURN] key - saves changes to the active field provided there is a valid entr', () => {
        cy.hover(d`headerCell-0-1`);
        cy.shouldBeVisible(d`visitNameBox`).click();
        cy.shouldBeVisible(d`visitNameInput`)
          .clear()
          .type('Batman{enter}');
        cy.hover(d`headerCell-0-2`);
        cy.shouldBeVisible(d`headerCell-0-1`).contains('Batman');
      });

      it('uses [ESC] key - to cancel the edit', () => {
        cy.hover(d`headerCell-0-1`);
        cy.shouldBeVisible(d`visitNameBox`).click();
        cy.shouldBeVisible(d`visitNameInput`)
          .clear()
          .type('Robin{esc}');
        cy.hover(d`headerCell-0-2`);
        cy.shouldBeVisible(d`headerCell-0-1`).contains('Batman');
      });

      it('checks if clicking outside while having unfinished business(editing error) should throw a toast notification, highlights the field in red border, and shaking animation', () => {
        cy.hover(d`headerCell-0-1`);
        cy.shouldBeVisible(d`visitNameBox`).click();
        cy.shouldBeVisible(d`visitNameInput`)
          .clear()
          .type('.*))9099');

        // save snapshot of border color that is blue, according to today.
        cy.get(d`visitNameBox`).getStyleSnapshot('borderColor');
        // random coords
        cy.get('body').click(35, 10);

        cy.get(d`headerCell-0-1`).should('have.class', 'animate__shakeX');
      });
    });
  },
  options,
);
