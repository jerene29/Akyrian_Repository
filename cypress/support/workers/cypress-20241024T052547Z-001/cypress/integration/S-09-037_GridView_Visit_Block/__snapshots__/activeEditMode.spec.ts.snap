exports[`S-09-037 > Active Edit Mode on Visit > keyboard shortcut > checks if clicking outside while having unfinished business(editing error) should throw a toast notification, highlights the field in red border, and shaking animation #0`] =
{
  "borderColor": "rgb(255, 0, 0)"
};
